posydon/popsyn/synthetic_population.py

```diff
@@ -197,36 +226,56 @@ def parse(self, S1_state=None, S2_state=None, binary_state=None,
             Read the POSYDON binary population in chuncks to prevent OFM error.
 
         """
+        
+        # if the user provided a single string instead of a list of strings
+        if type(path_to_data) is str and ('.h5' in path_to_data):
+            path_to_data = [path_to_data]
+        
+        # catch the case where the user did not provide a path to data     
+        if not (isinstance(path_to_data, list) and '.h5' in path_to_data[0]):
```

Comment:
Why do you only check for the first one? The second one could be a non h5 file. Thus I'd split the check on list and a check in a loop through the list.
Additionally, you should use `os.path.splitext(path_to_data[0])[-1]` to check the file extension.